### DISCUSSION TITLE
$this->withValidation doesn't work

### DISCUSSION THREAD

**Discussion by natanaelsinisalo** [livewire/livewire#4214] [1. Help] [answered]
I tried to access the validator instance ([following this documentation under the "Access Validator instance" section](https://laravel-livewire.com/docs/2.x/input-validation)) so I can trigger a toast notification after getting errors in validation.

I tried:

```php
use Illuminate\Validation\Validator;

class Example extends Component
{
    public function save()
    {
        $this->withValidator(function (Validator $validator) {
            $validator->after(function ($validator) {
                $this->toast('Please check errors in the form');
            });
        })->validate();
    }
}
```

and was greeted with an error:

`Method App\Http\Livewire\Example::withValidator does not exist. `

Tried to find something related to `withValidator` inside the Livewire `ValidatesInput` trait but didn't find anything.

Is this a bug? Or something that was deprecated but still in the docs?

**Comment by zepfietje** [comment, marked as answer]
Hey @natanaelsinisalo, not sure why it's not working, but I did find the following though: https://github.com/livewire/livewire/blob/6fb5d76e24a278de23c0a829e11e8073c11c2d2d/src/ComponentConcerns/ValidatesInput.php#L158

**Reply by natanaelsinisalo** [reply]
Oh, it's actually added in 2.7 and I'm currently running 2.6.X. That might explain why I wasn't able to use it 😅 Thanks!

**Reply by zepfietje** [reply]
Hah, glad that fixed it!

**Comment by Penderis** [comment]
Thank you for this, although I am on V3 the docs only show this being loaded in a boot method which does not work at all. The key to making this work for me is piping it to the validate() method which frankly makes sense but is not at all noted.
